test/python/WMComponent_t/AgentStatusWatcher_t/AgentStatusWatcher_t.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+"""
+AgentStatusWatcher test
+"""
+
+from __future__ import print_function
+import threading
+import unittest
+from WMQuality.TestInitCouchApp import TestInitCouchApp as TestInit
+
+from WMComponent.AgentStatusWatcher.AgentStatusPoller import AgentStatusPoller
+
+
+class AgentStatusWatcher_t(unittest.TestCase):
+    """
+    TestCase for WorkQueueManagerTest module
+    """
+
+    def setUp(self):
+        """
+        setup for test.
+        """
+        self.testInit = TestInit(__file__)
+        self.testInit.setLogging()
+        self.testInit.setDatabaseConnection()
+        #self.testInit.setSchema(customModules=["WMCore.WMBS"],
+        #                        useDefault=False)
+        #self.testInit.setSchema(customModules=["WMComponent.DBS3Buffer"],
+        #                        useDefault=False)
+        self.reqmonDBName = "wmstat_t"
+        self.localDBName = "wmstat_t_local"
+        self.testInit.setupCouch(self.reqmonDBName, "WMStats")
+        self.testInit.setupCouch(self.localDBName, "WMStats")
+        self.testDir = self.testInit.generateWorkDir()
+        return
+
+    def tearDown(self):
+        """
+        Database deletion
+        """
+        threading.currentThread()
+
+        self.testInit.delWorkDir()
+        self.testInit.tearDownCouch()
+        self.testInit.clearDatabase()
+        return
+
+    def getConfig(self):
+        """
+        _createConfig_
+
+        General config file
+        """
+
+        couchURL = self.testInit.couchUrl
+        config = self.testInit.getConfiguration()
+        self.testInit.generateWorkDir(config)
+
+        config.section_("Agent")
+        config.Agent.hostName = "localhost"
+        config.Agent.contact = "test@test.com"
+        config.Agent.teamName = "testTeam"
+        config.Agent.agentName = "testAgentName"
+        config.Agent.useMsgService = False
+        config.Agent.useTrigger = False
+        config.Agent.useHeartbeat = False
+
+        config.component_("AgentStatusWatcher")
+        config.AgentStatusWatcher.namespace = "WMComponent.AgentStatusWatcher.AgentStatusWatcher"
+        config.AgentStatusWatcher.jsonFile = "/dev/null"
```

Comment:
I need to clean up this jsonFile thing. It was one of those temporary things that end up becoming permanent..